tests/phpunit/integration/Modules/Analytics_4/Reset_AudiencesTest.php

```diff
@@ -0,0 +1,311 @@
+
+			foreach ( $test_dismissed_items as $dismissed_item ) {
+				$this->dismissed_items->add( $dismissed_item );
+			}
+
+			$user_dismissed_items = $this->dismissed_items->get();
+			foreach ( $test_dismissed_items as $dismissed_item ) {
+				$this->assertTrue( array_key_exists( $dismissed_item, $user_dismissed_items ) );
+			}
+		}
+		$this->user_options->switch_user( $backup_user_id );
+
+		// Update the propertyID to trigger reset.
+		$this->analytics->get_settings()->merge(
+			array(
+				'propertyID' => 'UA-222222',
+			)
+		);
+
+		// Confirm the users dismissed items have been reset.
+		foreach ( $this->users as $user ) {
+			$this->user_options->switch_user( $user->ID );
+
+			$user_dismissed_items = $this->dismissed_items->get();
+			foreach ( $test_dismissed_items as $dismissed_item ) {
+				$this->assertFalse( array_key_exists( $dismissed_item, $user_dismissed_items ) );
+			}
+		}
+		$this->user_options->switch_user( $backup_user_id );
+	}
+
+	public function test_reset_audience_segmentation_configured_audience_settings_across_multiple_users() {
+
+		$default_user_audience_settings = array(
+			'configuredAudiences'                => null,
+			'isAudienceSegmentationWidgetHidden' => false,
+			'didSetAudiences'                    => false,
+		);
+
+		$activated_user_audience_settings = array(
+			'configuredAudiences'                => array(
+				'properties/12345678/audiences/12345',
+				'properties/12345678/audiences/67890',
+			),
+			'isAudienceSegmentationWidgetHidden' => true,
+			'didSetAudiences'                    => true,
+		);
+
+		// Set configured audiences for each user.
+		$backup_user_id = $this->user_options->get_user_id();
+		foreach ( $this->users as $user ) {
+			$this->user_options->switch_user( $user->ID );
+
+			$this->audience_settings->merge(
+				array(
+					'configuredAudiences'                => $activated_user_audience_settings['configuredAudiences'],
+					'isAudienceSegmentationWidgetHidden' => $activated_user_audience_settings['isAudienceSegmentationWidgetHidden'],
+				)
+			);
+
+			$audience_settings = $this->audience_settings->get();
+			foreach ( array_keys( $default_user_audience_settings ) as $key ) {
+				$this->assertEquals( $activated_user_audience_settings[ $key ], $audience_settings[ $key ] );
+			}
+		}
+		$this->user_options->switch_user( $backup_user_id );
+
+		// Update the propertyID to trigger reset.
+		$this->analytics->get_settings()->merge(
+			array(
+				'propertyID' => 'UA-222222',
+			)
+		);
+
+		// Confirm the users audience settings have been reset.
+		foreach ( $this->users as $user ) {
+			$this->user_options->switch_user( $user->ID );
+			$audience_settings = $this->audience_settings->get();
+			foreach ( array_keys( $default_user_audience_settings ) as $key ) {
+				// As 'configuredAudiences' is updated to null, this key is removed.
+				if ( 'configuredAudiences' === $key ) {
+					$this->assertFalse( array_key_exists( $key, $audience_settings ) );
+				} else {
```

Comment:
As explained above, I think we expect the `configuredAudiences` key to still exist with a value of `null`. So, we should not need this special case.

Ideally, when we use `$this->audience_settings->delete()` method instead of `reset`, the option is removed, thus resulting in `$this->audience_settings->get()` will automatically return the default values ([see reference](https://github.com/google/site-kit-wp/blob/d6822a99a6ba0e7c8bc4581528eeb2b410cbeb7f/includes/Core/Storage/User_Setting.php#L87-L89)).